### DISCUSSION TITLE
Scheduleding a post for a later time

### DISCUSSION THREAD

**Discussion by marianunez** [mattermost/community#64] [Practical Ideas]
It would be great if we had a "Send Later" option to send the post later, maybe select a specific time or an event like the user is active.

**Comment by chetanyakan** [comment]
For DMs, be able to select other user's timezone

**Reply by esethna** [reply]
Or visibility/awareness of their notification schedule

**Comment by esethna** [comment]
Interesting that this is such a popular request. Perhaps the ability to set scheduled work hours for DND would help this overall pain point?

**Reply by marianunez** [reply]
Scheduled DnD work hours would be great to have. 

However, when sending something to someone else you don't know how they have their notifications settings configured. If this person does not use the DnD feature or turn off notifications on their mobile, they still may get pinged off-hours. Agreed that is on them but is something you may think about before sending. 

Additionally, you may want to send a post purposely certain time later so it can be actionable at the time is received (maybe the next Monday instead of a Saturday) and not lost in the posts that was off-hours.

**Reply by Willyfrog** [reply]
dnd hours work for not disturbing others, but I see more use of the feature on planning an event like mattercon. The community team could probably prepare most of the posts in advance, so they don't need to care too much to do things in time.

this obviously needs the option to reeschedule and cancel.

**Comment by jespino** [comment]
I created it some time ago, now is getting dust, but can be undusted and used.

https://github.com/jespino/mattermost-plugin-messages-queue
